### ISSUE TITLE
New WG on cloud-native conventions for planetary data, initial proposal

### ISSUE THREAD

**Issue by AndrewAnnex** [planetarysoftware/TSC#146]
I am soliciting feedback on a general idea to form either a working group or strategic initiative around the broad topic of using the latest cloud-native technologies with planetary datasets to support the use and creation of Analysis Ready Data. Given the rise of cloud-optimized geotiffs, tiled map services, spatiotemporal asset catalogs, various terrain encoding methods (e.g. RGB tiles, cesium), and persistent issues and incompatibilities with adhoc coordinate reference systems used by the community, there is a real need to define common metadata in a centralized way that can be used and shared by the community to ensure compatibility between services and ease-of-use. 

To be absolutely clear, this group would not be defining new standards, but will help define a set of metadata to configure existing standards.

A preliminary list of problems in approximate order:
1. Selecting a set of IAU WKT2 projections and ellipsoids for planetary bodies (N/S polar projections and equirectangular) and utilities to convert between malformed names/ellipsoids that are used in the community.
2. Creating a set of Geodetic TIFF grids (GTG) for various planetary bodies and contributing them to https://github.com/OSGeo/PROJ-data/
3.  Defining a set of TileMatrixSet grids for various planetary bodies and projections (initial ideas at https://github.com/andrewAnnex/planetcantile)
4. Defining a common set of RGB (24bit) Topography encodings for various planetary bodies tuned to apriori resolutions (e.g. 1 meter, 1 cm, etc), using common RGB encoding methods and or space filling curves.
5. Best practices for vector tiles and COG encodings in general. 
6. Decisions on how to use STAC and STAC extensions properly for various planetary bodies

This is a big set of hairy problems, too much I think for me to do alone, but they are not chicken-and-egg problems. A knock against this proposal is that these problems have probably all been solved by one person or another, but just haven't been collected all together, in which case this could be solved quickly. I think that the Moon and Mars are the best bodies to start out with to help move this effort along. 

I propose that for now we discuss if this is better served as a working group or strategic initiative proposal and make a new issue following the formal procedures layed out in this repository, hopefully to be discussed further during the next meeting of the TSC. Technical merits/discussions about the problems I detail above can happen on the openplanetary slack or elsewhere.

**Comment by jlaura** [comment]
@AndrewAnnex I am super glad that you wrote this and are wanting to push it forward. This is an area that we've been working in a lot, so even just having a community of practice around these topics will be a huge leap forward. 

I would suggest that we get a meeting setup to discuss, identify some easy wins (like #6, just documentation), and then dig into some of the harder items (e.g., #3). We can do this as a working group under the TC of interested parties. I'd be up for anything post Thanksgiving.

On #3, have you thought about just contributing back to morecantile? That maintainer is awesome, and it would be nice to extend his few base tile schemes. This is something I have been wanting to dig into more (I have Europa and Mars tiler servers deployed via Titiler on AWS right now, so progress is underway).

**Comment by AndrewAnnex** [comment]
thanks @jlaura, I'll look at my calendar soon and post some options. For contributing to morecantile, that can certainly happen, but I wanted a code area (planetcantile, which is also a great name) to experiment with first as I wanted a more 1:1 relationship of TMS specs and well-known WKT projections (related to 1 & 2, naming conventions are already showing issues here). If sufficient progress is made there we can certainly contribute the config files back to morecantile. 

I also wrote up some of the progress I've made for 4 here, which includes a very early version of the spec for rgb encodings (will need a few extra things like versioning): https://gist.github.com/AndrewAnnex/2ee19b2afe789faf7c81bdd978a7ed7c#file-rgb_encode_mars-ipynb

**Comment by jlaura** [comment]
@AndrewAnnex That sounds really reasonable! I'll work on getting the tile matrices that I have pushed into planetcantile so that we have that space. 

We should loop in @thareUSGS for a discussion on the WKT2 definitions that he has been working on pushing too.

**Comment by AndrewAnnex** [comment]
Generally Wednesdays afternoons and Friday mornings work for me:
12/2/22 9am-2pm PST
12/7/22 12pm-3pm PST
12/9/22 9am-2pm PST
12/14/22 12pm-3pm PST
.... and so on

**Comment by jlaura** [comment]
How about first Friday of the month @9AM PST? First time to touch base would be 12/2 @9AM PST (10AM MST, etc.)

**Comment by AndrewAnnex** [comment]
Okay let's go with **10am PST** (works slightly better for me) on 12/2? I may be able to post or dm a webex link friendlier to USGS folks but can also join a teams link.

**Comment by thareUSGS** [comment]
@AndrewAnnex So I assume you mean 10am PST (11am MST)? That works for me. Currently, 10am MST on 12/2 is booked for me.

BTW, did you really run zonalstats on the "astroesri" DEM :-)? Seems like it worked, but I wonder how long it took? I have tried to dissuade folks from running stats on that test server, but if it worked... They are hosting it in-kind, so I didn't want a ton of hard-hitting processes on it.

**Comment by jlaura** [comment]
@AndrewAnnex Do you want to setup a zoom or webex meeting? I can do a teams meeting with a dial in number if that is preferable.

**Comment by AndrewAnnex** [comment]
@thareUSGS : yes 10 am PST, I've updated the comment to reflect that. Yes I did, don't plan to anymore/only did it once or twice so yeah noted. It didn't take all that long though, maybe a minute or two but I wasn't timing it, but yeah I get what you are saying.

@jlaura I can set up a zoom link that I'll post here shortly.

**Comment by thareUSGS** [comment]
If it only took that long, the server probably did the optimized thing and gave you a subsampled (pyramided) version of the data. Meaning it didn't likely go down to full-resolution for the stats (for better or worse). To do it at pixel level, depending on rio, you would likely need to download the original, remove any pyramids and re-run.

**Comment by AndrewAnnex** [comment]
@thareUSGS it wasn't supposed to be super precise in the end, something we can discuss on Friday, but I don't think it is necessary to have per-MC tile encodings in the end. 

zoom meeting info:
https://caltech.zoom.us/j/83202438326

**Comment by jlaura** [comment]
This is spiraling into some other discussion. Having said that, if you want zonal stats and the input data are stac, it might be good to use stackstac and a compute cluster. Then the xarray/dask magic can happen in parallel. Maybe another reason that that stac ecosystem is so sweet!

**Comment by AndrewAnnex** [comment]
yeah it kinda is, certainly a thing that can be discussed as part of the discussion of bullet point number 6 above, though for the purposes of bullet point number 4 (reason for the zonal stats I've done in the notebook linked above) it's not really important at the moment.

**Comment by AndrewAnnex** [comment]
After looking into this a bit more, while I think a WG could grow out of this effort it would be more time efficient just to call this a strategic initiative that I could lead alone or with @jlaura if he's interested. Since for the moment it is just me and Jay primarily working on the tasks I don't think the formalities of a wg are needed until there are more active participants. 

If there are no objections @rbeyer I'll open a pr for that, but will wait a few days for others to weigh in. (once that pr is merged, or another decision is reached this issue will be closed)

**Comment by jlaura** [comment]
I think a strategic initiative is the way to go. We'll have a standing, 'please report' in the agenda, but no other real infrastructure needed. This probably makes sense until the point that spawn off some sort of planetary cloud standards propagation WG that is tackling things a little less ad hoc (ad hoc is good right now since I don't know that we even have a strong handle on the landscape yet!).
